packages/api/src/services/event-service/event.service.ts

```diff
@@ -0,0 +1,89 @@
+import {Event} from '../../models/events';
+import {Service} from '../../providers/service/service';
+import {ObjectUtil} from '../utils';
+
+/**
+ * Represents an observer for events, encapsulating a callback to handle event notifications.
+ *
+ * @template T - The type of the event payload.
+ */
+export class EventObserver<T> {
+
+  /**
+   * A callback function to notify the observer of an event.
+   *
+   * @param eventPayload - The payload of the event to notify the observer about.
+   */
+  notify: (eventPayload: T) => void;
+
+  constructor(callback: (payload: T) => void) {
+    this.notify = callback;
+  }
+}
+
+/**
+ * A service that manages event subscriptions and emissions.
+ *
+ * Allows components or modules to subscribe to specific events and be notified when those events are emitted.
+ */
+export class EventService implements Service {
+
+  /**
+   * A map of event names to their corresponding list of subscribers.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private eventSubscribers: Record<string, EventObserver<any>[]> = {};
```

Comment:
Can we use a WeakMap here

Reply:
Added logic that removes the key if the value array is empty